dcos_installer/config_util.py

```diff
@@ -23,6 +23,8 @@ def do_configure(config):
     # Write some package metadata
     pkgpanda.util.write_json(CLUSTER_PACKAGES_PATH, gen_out.cluster_packages)
 
+    return gen_out
```

Comment:
This looks like an opportunity for a refactor. Instead of calling `gen.generate()` at the top of the function and returning its `gen_out` at the bottom, we could decompose this a bit and separate config generation from populating `genconf/serve/`.

Quick sketch:

```
def onprem_generate(config):
    return gen.generate(config.as_gen_format(), extra_sources=[gen.build_deploy.bash.onprem_source])

def make_serve_dir(gen_out):
    subprocess.check_call(['mkdir', '-p', SERVE_DIR])
    gen.build_deploy.bash.generate(gen_out, SERVE_DIR)
    # etc.

# You can do this:
make_serve_dir(onprem_generate(config))

# Or if you need to reuse gen_out, you can do this:
gen_out = onprem_generate(config)
make_serve_dir(gen_out)
do_something_else_with(gen_out)
```